pkg/proxy/backend/authenticator.go

```diff
@@ -247,44 +247,44 @@ func forwardMsg(srcIO, destIO *pnet.PacketIO) (data []byte, err error) {
 	return
 }
 
-func (auth *Authenticator) handshakeSecondTime(logger *zap.Logger, clientIO, backendIO *pnet.PacketIO, backendTLSConfig *tls.Config, sessionToken string) error {
+func (auth *Authenticator) handshakeSecondTime(logger *zap.Logger, clientIO, backendIO *pnet.PacketIO, backendTLSConfig *tls.Config, sessionToken string) (uint64, error) {
 	if len(sessionToken) == 0 {
-		return errors.New("session token is empty")
+		return 0, errors.New("session token is empty")
 	}
 
 	// write proxy header
 	if err := auth.writeProxyProtocol(clientIO, backendIO); err != nil {
-		return err
+		return 0, err
 	}
 
-	_, backendCapability, err := auth.readInitialHandshake(backendIO)
+	_, backendCapability, connID, err := auth.readInitialHandshake(backendIO)
 	if err != nil {
-		return err
+		return connID, err
 	}
 
 	if err := auth.verifyBackendCaps(logger, pnet.Capability(backendCapability)); err != nil {
-		return err
+		return connID, err
 	}
 
 	if err = auth.writeAuthHandshake(
 		backendIO, backendTLSConfig, backendCapability,
 		pnet.AuthTiDBSessionToken, hack.Slice(sessionToken), pnet.ClientPluginAuth,
 	); err != nil {
-		return err
+		return connID, err
 	}
 
-	return auth.handleSecondAuthResult(backendIO)
+	return connID, auth.handleSecondAuthResult(backendIO)
 }
 
-func (auth *Authenticator) readInitialHandshake(backendIO *pnet.PacketIO) (serverPkt []byte, capability pnet.Capability, err error) {
+func (auth *Authenticator) readInitialHandshake(backendIO *pnet.PacketIO) (serverPkt []byte, capability pnet.Capability, connID uint64, err error) {
 	if serverPkt, err = backendIO.ReadPacket(); err != nil {
 		return
 	}
 	if pnet.IsErrorPacket(serverPkt) {
 		err = pnet.ParseErrorPacket(serverPkt)
 		return
 	}
-	capability, _, _ = pnet.ParseInitialHandshake(serverPkt)
+	capability, connID, _ = pnet.ParseInitialHandshake(serverPkt)
```

Comment:
TiDB connection ID is actually 64 bits: https://github.com/pingcap/tidb/blob/master/util/globalconn/globalconn.go#L39
But TiDB only sends 32 bits, which is limited by the MySQL protocol: https://github.com/pingcap/tidb/blob/master/server/conn.go#L380
All the TiDB logs use the 64-bit conn ID, so it's useless to log the 32-bit ID in the TiProxy log.
You can only send `select connection_id()` after handshake to query the actual connection ID.